### ISSUE TITLE
Window doesn't get tiled to the left (or right) of the screen when you press Meta+← (or Meta+→) while the window is maximized

### ISSUE THREAD

**Issue by JohanLorenzo** [seleznev/firefox-extension-htitle#57]
Hi,

Thank you for this great addon which enables me to save some room on my screen! I found this case which might be worth to be called out.

## Steps to reproduce
1. Maximize the window (like with Meta+↑)
1. Tile it to the left with Meta+←

## Expected results
Like with any other Gnome application, the window should be tiled to the left.

## Actual results
The window gets un-maximized but not tiled anywhere. You have to use the tiling shortcut again to make it happen.

**Comment by plsng** [comment]
Works for me (tm).

**Comment by JohanLorenzo** [comment]
Here's [a short screencast](http://mzl.la/1E9yrq2) to highlight the issue. I wanted to tile the window to the right and it went to the left of the screen (without being tiled).

After a bit of investigation, I notice that after switching "Hide title" from "When maximized" to "Always" and then back to "When maximized", the issue disappeared. The issue might be different than the one I described above. I'll try to get more information.

**Comment by JohanLorenzo** [comment]
I am not able to reproduce the problem at all when "Hide title" is set to "Always". But I can run into some issue if it's set to "When maximized". I am currently in a state where the title bar is not hidden anymore. I'm not sure of the steps to reproduce here. Is it possible to get some logs out?

**Comment by seleznev** [comment]
@JohanLorenzo, I can reproduce it with enabled "Legacy mode" checkbox, but not without. But for legacy mode it's correct behavior. Can you disable this checkbox?

> Is it possible to get some logs out?

You can try to set `extensions.htitle.debug` to true and check the Browser console. Not sure that it can help, but can help to understand what HTitle does in this time.

**Comment by JohanLorenzo** [comment]
Legacy mode might be the issue. It seems that the extensions has some trouble to know if I have to use legacy mode or not. In other words, if:

1. I open about:addons and then the preferences of htitle, I can enable/disable legacy mode
2. And then if I open a new tab with about:addons and the preferences of htitle, I get the message
`You must have "libX11.so" and "libgtk-x11-2.0.so libs in your system to disable it.`

Is this an expected behavior @seleznev ?

**Comment by seleznev** [comment]
> Is this an expected behavior?

No, it isn't. You should see this messages at 1. step too. I have no idea what happened. If you will provide details steps to reproduce (with clean profile), I can to try fix it.
